### PR TITLE
Bus and branch result extension

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
We cannot add extension to BusResult, BranchResult and ThreeWindingsTransformerResult.

**What is the new behavior (if this is a feature change)?**
We can

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*
This is a breaking change because now we need to use custom jackson serializer (to handle the extension part) to have same json as before and because BusResults has been renamed to BusResult to be consistent with the others.

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
Sonar fail because of code duplication. I don't think we can/should fix it as it is not really possible/reasonable to share code for json parsing.

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/BusResultSerializer.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+public class BusResultSerializer extends StdSerializer<BusResult>  {
+
+    public BusResultSerializer() {
+        super(BusResult.class);
+    }
+
+    @Override
+    public void serialize(BusResult busResult, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException {
+        jsonGenerator.writeStartObject();
+        jsonGenerator.writeStringField("voltageLevelId", busResult.getVoltageLevelId());
```
Comment (miovd) [inline security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/BusResultSerializer.java:29]: Maybe not to be completely handled in this PR, but shouldn't these results be versionable? Maybe you could only add `jsonGenerator.writeStringField("version", "1.0");` for now and handle it properly if we add a change

**Review by miovd — COMMENTED** [review]
I mostly agree with this PR, the note I made for `BusResult` is valid for the other serializers

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/BusResultSerializer.java (hunk 1 above)
Comment (geofjamg) [reply security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/BusResultSerializer.java:29]: Bus/branch/3wt results are part of security analysis result which is already versioned.

Comment (miovd) [reply security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/BusResultSerializer.java:29]: Indeed, you're right... Okay for me then!

**Review by miovd — APPROVED** [review]

**Pull request merged**